logger.go

```diff
@@ -5,20 +5,22 @@ import (
 	"errors"
 	"time"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus"
 	"gorm.io/gorm"
 	gormlogger "gorm.io/gorm/logger"
 	"gorm.io/gorm/utils"
 )
 
 type logger struct {
+	log                   *logrus.Logger
 	SlowThreshold         time.Duration
 	SourceField           string
 	SkipErrRecordNotFound bool
 }
 
-func New() *logger {
+func New(l *logrus.Logger) *logger {
```

Comment:
What do you think about changing it to be an optional attribute to avoid breaking changes to the existing interface?

So either just pass a `Nil` value and check for "nilness" in the constructor or use the optional pattern to be able to pass additional options in the future: https://www.sohamkamani.com/golang/options-pattern/
This way it would also be possible to keep the constructor as it is without the need to pass a `Nil` value ;)